openwrt_api.py

```diff
@@ -0,0 +1,400 @@
+        time in milliseconds
+    `avgping`: the average round trip ping time in milliseconds
+    `maxping`: the maximum (slowest) round trip ping time in
+        milliseconds
+    `jitter`: the standard deviation between round trip ping times
+        in milliseconds
+  """
+  if type(dest_ip) is not str:
+    raise RepyArgumentError("Provided host must be a string!")
+
+  if type(count) is not int:
+    raise RepyArgumentError("Provided count must be an int!")
+
+  # Check the input arguments (sanity)
+  if not emulcomm._is_valid_ip_address(dest_ip):
+    raise RepyArgumentError("Provided dest_ip is not valid! IP: '" + dest_ip + "'")
+
+  if count < 1:
+    raise RepyArgumentError("Provided count must be more than 0")
+
+  process = portable_popen.Popen(['ping', '-c', str(count), dest_ip],)
+  ping_output, err = process.communicate()
+  return _parse(str(ping_output))
+
+def get_station(interface):
+  """
+   <Purpose>
+      To get station statistic information such as the amount of tx/rx bytes, the 
+      last TX bitrate (including MCS rate) you can do
+
+   <Arguments>
+      interface: the name of the interface to gather network information.
+
+   <Exceptions>
+      RepyArgumentError is raised if the argument isn't valid types or values
+
+   <Side Effects>
+      None
+
+   <Returns>
+     station statistic informatio as a dict such as {{'signal': '-77 [-78, -83] dBm', 
+    'station': '54:4e:90:06:f3:6f', 'tx bitrate': '57.8 MBit/s MCS 5 short GI', 'rx 
+    bitrate': '1.0 MBit/s', 'signal_avg': '-88 [-90, -92] dBm'}, {'signal': '-40 [-42, 
+    -44] dBm', 'station': 'ac:81:12:53:8e:0f', 'tx bitrate': '65.0 MBit/s MCS 6 short 
+    GI', 'rx bitrate': '18.0 MBit/s', 'signal_avg': '-42 [-44, -48] dBm'}}
+
+  """
+  if type(interface) is not str:
+    raise RepyArgumentError("get_station() takes a string as argument.")
+
+  # Check the input arguments (sanity)
+  if interface not in get_network_interface():
+    raise RepyArgumentError("interface "+ interface + " is not available.")
+
+  iw_process = portable_popen.Popen(['iw', 'dev',interface, 'station','dump'])
+  iw_output, _ = iw_process.communicate()
+  iw_lines = textops.textops_rawtexttolines(iw_output)
+
+  station = textops.textops_grep("Station", iw_lines)
+  station = textops.textops_cut(station, delimiter=" ", fields=[1])
+
+  signal = textops.textops_grep("signal:", iw_lines)
+  signal = textops.textops_cut(signal, delimiter=":", fields=[1])         
+
+  signal_avg = textops.textops_grep("signal avg:", iw_lines)
+  signal_avg = textops.textops_cut(signal_avg, delimiter=":", fields=[1])         
+
+  tx_bitrate = textops.textops_grep("tx bitrate:", iw_lines)
+  tx_bitrate = textops.textops_cut(tx_bitrate, delimiter=":", fields=[1])         
+
+  rx_bitrate = textops.textops_grep("rx bitrate:", iw_lines)
+  rx_bitrate = textops.textops_cut(rx_bitrate, delimiter=":", fields=[1])         
+
+  result = []
+
+  for i in range(len(station)):
```

Comment:
Hmmm, so all os the grepped/cut lists are of identical length? Then you probably want to loop like this:
```
  for st, sig, sig_avg, tx, rx in  zip(station, signal, signal_avg, tx_bitrate, rx_bitrate):
    ....
```

(`zip`ping lists makes a list of tuples containing the 0th, 1st, 2nd, etc. entry of each of the lists, respectively. Then, we can unpack from the current tuple into multiple variables in the loop-local scope.)